Announcement blogpost for the 1.15 release (#526)
apache/flink-web: _posts/2022-04-26-1.15-announcement.md

---
layout: post
title:  "Announcing the Release of Apache Flink 1.15"
subtitle: ""
date: 2022-04-26T08:00:00.000Z
categories: news
authors:
- yungao:
  name: "Yun Gao"
  twitter: "YunGao16"
- joemoe:
  name: "Joe Moser"
  twitter: "JoemoeAT"

---

Thanks to our well-organized and open community, Apache Flink continues 
[to grow](https://www.apache.org/foundation/docs/FY2021AnnualReport.pdf) as a 
technology and remain one of the most active projects in
the Apache community. With the release of Flink 1.15, we are proud to announce a number of 
exciting changes.

> One of the main concepts that makes Apache Flink stand out is the unification of 
> batch (aka bounded) and stream (aka unbounded) data processing. A lot of 

Reviewer: ```suggestion
One of the main concepts that makes Apache Flink stand out is the unification of 
batch (aka bounded) and stream (aka unbounded) data processing, which helps reduce the complexity of development. A lot of 
```

effort went into this unification in the previous releases but you can expect more efforts in this direction. 

> Apache Flink is not only growing when it comes to contributions and users, but

Reviewer: ```suggestion

Apache Flink is not only growing when it comes to contributions and users, but
```

also out of the original use cases. We are seeing a trend towards more business/analytics 
use cases implemented in low-/no-code. Flink SQL is the feature in the Flink ecosystem 
that enables such uses cases and this is why its popularity continues to grow.  

Apache Flink is an essential building block in data pipelines/architectures and 
is used with many other technologies in order to drive all sorts of use cases. While new ideas/products
may appear in this domain, existing technologies continue to establish themselves as standards for solving 
mission-critical problems. Knowing that we have such a wide reach and play a role in the success of many 

> projects, it is important that the experience of 
> integrating with Apache Flink is as seamless and easy as possible. 

Reviewer: ```suggestion
projects, it is important that the experience of 
integrating Apache Flink with cloud infrastructures and these systems is as seamless and easy as possible. 
```

   By keeping the size consistently small, checkpointing time becomes more predictable.
3. Less work on recovery: with more frequently checkpoints are taken, less data need 
   to be re-processed after each recovery.

The changelog state backend helps achieve the above by continuously 
persisting state changes on non-volatile storage while performing state materialization 
in the background.

### Repeatable cleanup

In previous releases of Flink, cleaning up job-related artifacts was done only once which might have resulted in abandoned artifacts in case of an error. In this version, Flink will try to run the cleanup again to avoid leaving artifacts behind. This retry mechanism runs until it was successful, by default. Users can change this behavior by configuring the [repeatable cleanup options](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/config/#retryable-cleanup). Disabling the retry strategy will lead to Flink behaving like in previous releases.

There is still work in progress around cleaning up checkpoints, which is covered by 
[FLINK-26606](https://issues.apache.org/jira/browse/FLINK-26606).

### OpenAPI

> Flink is now providing the REST API specifications following the 
> [OpenAPI](https://www.openapis.org) standard.
> This allows the REST API to be used with standard tools that are implementing the
> OpenAPI standard.
> You can find the specifications [here](https://nightlies.apache.org/flink/flink-docs-master/docs/ops/rest_api/#jobmanager).

Reviewer 2: ```suggestion
Flink is now providing an experimental REST API specification following the 
[OpenAPI](https://www.openapis.org) standard.
This allows the REST API to be used with standard tools that are implementing the
OpenAPI standard.
You can find the specification [here](https://nightlies.apache.org/flink/flink-docs-master/docs/ops/rest_api/#jobmanager).
```

batch processing, became more accessible through improvements in the SQL components, 
and now integrates better with other technologies. 

It is also worth mentioning that 
the community has set up a new home for 
the [CDC connectors](https://ververica.github.io/flink-cdc-connectors/release-2.1/index.html), 
the [connector repository](https://cwiki.apache.org/confluence/display/FLINK/Connectors) 
will be externalized 
([with the Elasticsearch sink as a first example](https://github.com/apache/flink-connector-elasticsearch/)), 
and there is now a 
[Kubernetes operator](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/) 
([announcement blogpost](https://flink.apache.org/news/2022/04/03/release-kubernetes-operator-0.1.0.html) 
maintained by the community.

Moving forward, the community will continue to focus on making Apache Flink a true 
unified stream and batch processor and work on better integrating Flink into the cloud-native 
ecosystem. 

## Upgrade Notes

While we aim to make upgrades as smooth as possible, some of the changes require users 
to adjust some parts of the program when upgrading to Apache Flink 1.15. Please take a 
look at the

> [release notes](https://nightlies.apache.org/flink/flink-docs-release-1.14/release-notes/flink-1.14/)

Reviewer: Hi @joemoe  Here should be `https://nightlies.apache.org/flink/flink-docs-release-1.15/release-notes/flink-1.15/`
